### PR TITLE
Add number formatting

### PR DESCRIPTION
**Pull request by seemyvest**
As discussed in #13. 

I ended up splitting it into two macros, one for integers and one for floats. I also spilt the tests into three macros for unsigned and signed integers, and floats.

Let me know if you have any comments 👍

### CODE REVIEW

File: src/fmtnum.rs [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+                }
+
+                if self.alternate() {
+                    match ty {
+                        'b' => self.write_str("0b").unwrap(),
+                        'o' => self.write_str("0o").unwrap(),
+                        'x' | 'X' => self.write_str("0x").unwrap(),
+                        _ => {
+                            let mut msg = String::new();
+                            write!(msg, "alternate ('#') cannot be used with type {:?}", ty).unwrap();
```
Comment (vitiral) [inline src/fmtnum.rs]: when I first wrote this lib I didn't know about `format!()`. Please use that instead here. You can even get rid of `msg` variable alltogether.

File: src/lib.rs [hunk 2]
```diff
@@ -10,13 +10,17 @@ mod tests;
...
+
+#[macro_use]
+mod fmtnum;
 
 pub use types::{Result, FmtError, Alignment, Sign};
 pub use fmtstr::strfmt_map;
 pub use formatter::Formatter;
 
+// u128 & i128 unstable (see https://github.com/rust-lang/rust/issues/35118)
+fmtint!(u8 i8 u16 i16 u32 i32 u64 i64 usize isize);
```
Comment (vitiral) [inline src/lib.rs]: whoa! I didn't know this was possible, I thought they would have to each be their own line. Good work!

**Comment by vitiral** [comment]
Looks good, from [travis run]():
```
test tests::strfmt::test_i8 ... ok
test tests::strfmt::test_i16 ... ok
test tests::strfmt::test_i32 ... ok
test tests::strfmt::test_i64 ... ok
test tests::strfmt::test_isize ... ok
test tests::strfmt::test_u8 ... ok
test tests::strfmt::test_u16 ... ok
test tests::strfmt::test_u32 ... ok
test tests::strfmt::test_u64 ... ok
test tests::strfmt::test_usize ... ok
test tests::strfmt::test_f32 ... ok
test tests::strfmt::test_f64 ... ok
test tests::strfmt::test_ints_basic ... ok
```

File: src/fmtnum.rs (hunk 1 above)
Comment (vitiral) [reply src/fmtnum.rs]: whoops, I'll handle it in a separate commit... maybe (who really cares, haha).

Thanks for the PR!

**Comment by seemyvest** [comment]
No problem 👍

**Comment by vitiral** [comment]
I just publised `v0.1.6` with these changes, thanks!

**Pull request merged**